internal/output.go

```diff
@@ -229,3 +246,39 @@ func coverageParse(cov string) float64 {
 func coverageColor(cov float64) string {
 	return ifelse(cov < 50, "red", ifelse(cov < 75, "yellow", "green"))
 }
+
+// If coverProfile is set, calculate accurately, otherwise just average coverages
+func getTotalCoverage(coverProfile string, coverages []float64) (float64, bool) {
+	cpExists := coverProfile != "" && fileExists(coverProfile)
+
+	if !cpExists {
+		return sliceAvg(coverages), true
+	}
+
+	var cov float64
+	var stat float64
+
+	readFile, err := os.Open(coverProfile)
+	if err != nil {
+		fmt.Println("Error: failed to read cover profile")
+		return sliceAvg(coverages), true
+	}
+	defer readFile.Close()
+
+	fileScanner := bufio.NewScanner(readFile)
+	fileScanner.Split(bufio.ScanLines)
+
+	for fileScanner.Scan() {
+		line := fileScanner.Text()
+		parts := strings.Split(line, " ")
```

Comment:
Following cover calculation is based on the `awk` command at
https://eleni.blog/2021/01/24/deep-diving-in-the-go-coverage-profile/

```
go test --count=1 -coverprofile=coverage.out ./... ; \
cat coverage.out | \
awk 'BEGIN {cov=0; stat=0;} \
	$3!="" { cov+=($3==1?$2:0); stat+=$2; } \
    END {printf("Total coverage: %.2f%% of statements\n", (cov/stat)*100);}'
```